[BinaryPlatforms]: Flesh out CPUID support

staticfloat opened the pull request:
This PR fleshes out (and slightly renames) some pieces of the CPUID
machinery to better support BB's multi-march system.  It declares at
least one valid microarchitecture for all architectures, and splits the
CPUID module out into its own source file for readability.

staticfloat committed:
[BinaryPlatforms]: Flesh out CPUID support

This PR fleshes out (and slightly renames) some pieces of the CPUID
machinery to better support BB's multi-march system.  It declares at
least one valid microarchitecture for all architectures, and splits the
CPUID module out into its own source file for readability.

staticfloat commented:
@giordano just FYI

@vchuravy I guess we have no C CPUID code on power yet?  Too bad, we'll just compile everything for power8 for now then, until Julia gains the ability to select higher forms of computation.

staticfloat committed:
[BinaryPlatforms]: Add some normalization to `Sys.ARCH` values

Get rid of `cpu_family()`, it's kind of confusingly named and unnecessary.

giordano commented on base/cpuid.jl:
```diff
@@ -63,8 +63,30 @@ const ISAs_by_family = Dict(
+        arch = "armv6l"
+    elseif arch ∈ ("arm", "armv7", "armv8", "armv8l")
+        arch = "armv7l"
+    elseif arch ∈ ("arm64")
```
```suggestion
    elseif arch ∈ ("arm64",)
```

staticfloat committed:
[BinaryPlatforms]: Add `Base.show()` override for `Platform`

This should be a little more friendly to those just looking to inspect
some of these objects

staticfloat committed:
[BinaryPlatforms]: Fix `triplet()` representation with `os_version`
